### PR TITLE
NATIVE-66: Add FetcherModule for ios

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

**Comment by maxammann** [comment]
rdy for review

File: android/app/src/main/java/com/integreat/fetcher/FetchResult.java [hunk 1]
```diff
@@ -1,16 +1,16 @@
 package com.integreat.fetcher;
 
-import java.time.ZonedDateTime;
+import java.util.Date;
```
Comment (Schedulaar) [inline android/app/src/main/java/com/integreat/fetcher/FetchResult.java]: why did you revert this to date?

File: android/app/src/main/java/com/integreat/fetcher/FetcherModule.java [hunk 2]
```diff
@@ -37,6 +37,8 @@ public void fetchAsync(final ReadableMap urls, final Promise promise) {
                 expectedFetchCount, promise
         );
 
+
+
```
Comment (Schedulaar) [inline android/app/src/main/java/com/integreat/fetcher/FetcherModule.java]: Remove (or dont) these two blank lines.

File: src/modules/endpoint/sagas/fetchResourceCache.js [hunk 3]
```diff
@@ -38,11 +37,8 @@ export default function * fetchResourceCache (
 
   try {
     const targetUrls = mapValues(fetchMap, ([url]) => url)
-    if (Platform.OS !== 'android') {
-      return
-    }
 
-    const results: FetchResultType = yield call(new FetcherModule().fetchAsync, targetUrls, progress => {})
+    const results: FetchResultType = yield call(new FetcherModule().fetchAsync, targetUrls, progress => console.log(progress))
```
Comment (Schedulaar) [inline src/modules/endpoint/sagas/fetchResourceCache.js]: remove (or not) console.log

**Comment by Schedulaar** [comment]
I guess you can remove the `wip` tag?

File: android/app/src/main/java/com/integreat/fetcher/FetchResult.java (hunk 1 above)
Comment (maxammann) [reply android/app/src/main/java/com/integreat/fetcher/FetchResult.java]: ZonedDateTime depends on java 8 which is only available above api level 26

**Comment by maxammann** [comment]
@klinzo your review is missing here

File: android/app/src/main/java/com/integreat/fetcher/FetchResultCollector.java [hunk 4]
```diff
@@ -23,16 +21,21 @@
...
+
+        TimeZone timezone = TimeZone.getTimeZone("UTC");
+        iso8601Format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'", Locale.ENGLISH);
+        iso8601Format.setTimeZone(timezone);
     }
 
     @Override
     public synchronized void failed(String url, File targetFile, String message) {
-        fetchResults.put(targetFile.getAbsolutePath(), new FetchResult(url, ZonedDateTime.now(ZoneOffset.UTC), false, message));
+        fetchResults.put(targetFile.getAbsolutePath(), new FetchResult(url, new Date(), false, message));
```
[the remark this answers is not in the document]
Comment (maxammann) [reply android/app/src/main/java/com/integreat/fetcher/FetchResultCollector.java]: `Date` does not contain information about the timezone so there is no need to specify UTC

**Pull request merged**